app/assets/javascripts/darkswarm/directives/open_street_map.js.coffee

```diff
@@ -61,28 +33,47 @@ Darkswarm.directive 'ofnOpenStreetMap', ($window, Enterprises, EnterpriseModal,
 
     displayMap = ->
       setMapDimensions()
-      averageLatitude = averageAngle("latitude")
-      averageLongitude = averageAngle("longitude")
       zoomLevel = 6
       map = L.map('open-street-map')
       L.tileLayer.provider(openStreetMapProviderName, openStreetMapProviderOptions).addTo(map)
-      map.setView([averageLatitude, averageLongitude], zoomLevel)
+      map.setView([initialLatitude(), initialLongitude()], zoomLevel)
```

Comment:
These two methods are only called in this one place, right? I think I would extract `#initialLatitude` and `#initialLongitude` (and possibly `#geocodedEnterprises`) to the service as well, and just have something like this:
```js
map.setView([MapCentreCalculator.initialLatitude(), MapCentreCalculator.initialLongitude()], zoomLevel)
```

Reply:
@Matt-Yorkley yes that works well. I have moved those method into the `MapCentreCalculator` service now.